src/App.tsx

```diff
@@ -1,19 +1,158 @@
+/* eslint-disable no-console */
 import React from 'react';
-import './App.css';
+import './App.scss';
 
 import users from './api/users';
+import todos from './api/todos';
+import { Todo } from './types';
+import { TodoList } from './components/TodoList';
+import { prepareTodos } from './prepareTodos';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type State = {
+  todos: Todo[];
+  title: string;
+  userId: number;
+  isUserSelected: boolean;
+  isTitleWritten: boolean;
+  tooBigTitle: boolean;
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    todos,
+    title: '',
+    userId: 0,
+    isUserSelected: true,
+    isTitleWritten: true,
+    tooBigTitle: false,
+  };
+
+  handleOnSubmit = (event: React.SyntheticEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!this.validateSumbitData() || this.state.title.length > 60) {
+      return;
+    }
+
+    this.setState(currentState => {
+      const { userId, title, todos: todoList } = currentState;
+      const newTodo:Todo = {
+        userId,
+        title,
+        id: todoList[todoList.length - 1].id + 1,
+        completed: false,
+      };
+
+      return {
+        todos: [...currentState.todos, newTodo],
+      };
+    });
+
+    this.setState({
+      title: '',
+      userId: 0,
+    });
+  };
+
+  onChecked = (id:number) => {
+    this.setState((currentState) => {
+      const copy = [...currentState.todos];
+
+      copy[id - 1].completed = !copy[id - 1].completed;
```

Comment:
What if we add a removing button tomorrow? Bad practice to attach to index by id element